<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Systems level pathway analysis in disease • Single Cell Pathway Analysis (SCPA)</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Systems level pathway analysis in disease">
<meta property="og:description" content="SCPA">
<meta property="og:image" content="jackbibby1.github.io/SCPA/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-W1XR84YEPZ"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-W1XR84YEPZ');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Single Cell Pathway Analysis (SCPA)</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.5.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">General information</li>
    <li>
      <a href="../articles/interpreting_scpa_output.html">Interpreting the SCPA output</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Generic example</li>
    <li>
      <a href="../articles/quick_start.html">Quick start</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Speeding up SCPA</li>
    <li>
      <a href="../articles/parallel_implementation.html">Parallel processing for faster analysis</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Interacting with gene sets</li>
    <li>
      <a href="../articles/using_gene_sets.html">Using gene sets</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Specific examples</li>
    <li>
      <a href="../articles/comparing_two_populations.html">Comparing two populations</a>
    </li>
    <li>
      <a href="../articles/seurat_comparison.html">Seurat/SingleCellExperiment comparison</a>
    </li>
    <li>
      <a href="../articles/pseudotime.html">Comparing across pseudotime</a>
    </li>
    <li>
      <a href="../articles/systematic_tissue_comparison.html">Systematic comparisons of cells across tissue</a>
    </li>
    <li>
      <a href="../articles/disease_comparison.html">Systems level pathway analysis in disease</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Visualisation</li>
    <li>
      <a href="../articles/visualisation.html">Visualising SCPA output</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jackbibby1/SCPA/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Systems level pathway analysis in disease</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/jackbibby1/SCPA/blob/HEAD/vignettes/disease_comparison.Rmd" class="external-link"><code>vignettes/disease_comparison.Rmd</code></a></small>
      <div class="hidden name"><code>disease_comparison.Rmd</code></div>

    </div>

    
    
<p>Here we’re going to perform a systems level characterisation of
pathway perturbations in a disease. Applying a systems level analysis of
a disease using scRNA-seq data allows you to get a really good
understanding of a few important aspects:</p>
<ul>
<li>Which pathways show the greatest perturbation in the disease</li>
<li>Which cell types show the largest perturbation in their
transcriptional profile</li>
<li>Shared and unique pathway perturbations across cell types</li>
</ul>
<p>Because SCPA can be easily applied in a systems level way, it allows
for the identification of the most relevant pathways and cell types in
any given disease.</p>
<div class="section level2">
<h2 id="loading-in-datapackages">Loading in data/packages<a class="anchor" aria-label="anchor" href="#loading-in-datapackages"></a>
</h2>
<p>Let’s load in a few packages</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://jackbibby1.github.io/SCPA/">SCPA</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap" class="external-link">ComplexHeatmap</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/circlize" class="external-link">circlize</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span></code></pre></div>
<p>And load in a dataset from <a href="https://www.nature.com/articles/s41591-020-0944-y" class="external-link">Wilk, A…Blish,
C</a> that creates a single cell atlas of peripheral blood immune cells
in COVID-19 patients. You can download this dataset <a href="https://drive.google.com/file/d/1eYGW0psWFsxaRRtKTZCK6ESfPSQX0zeq/view?usp=sharing" class="external-link">here</a></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">blood_atlas</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"covid_blood_atlas.rds"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/Idents.html" class="external-link">Idents</a></span><span class="op">(</span><span class="va">blood_atlas</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">blood_atlas</span><span class="op">$</span><span class="va">cell.type</span></span></code></pre></div>
<p>Let’s have a quick look at the data, and get rid of any cell
populations that aren’t represented in both healthy and disease
datasets.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">blood_atlas</span>, label <span class="op">=</span> <span class="cn">T</span>, group.by <span class="op">=</span> <span class="st">"cell.type.fine"</span>, split.by <span class="op">=</span> <span class="st">"Status"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>aspect.ratio <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="../reference/figures/covid_atlas.png" style="width:100.0%"></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">idents</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Class-switched B"</span>, <span class="st">"IgG PB"</span>, <span class="st">"IgA PB"</span>, <span class="st">"CD14 Monocyte"</span>, <span class="st">"CD8m T"</span>,</span>
<span>            <span class="st">"CD4m T"</span>, <span class="st">"CD4n T"</span>, <span class="st">"B"</span>, <span class="st">"NK"</span>, <span class="st">"Neutrophil"</span>, <span class="st">"CD16 Monocyte"</span>, <span class="st">"gd T"</span>,</span>
<span>            <span class="st">"pDC"</span>, <span class="st">"SC &amp; Eosinophil"</span>, <span class="st">"DC"</span><span class="op">)</span></span>
<span><span class="va">blood_atlas</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">blood_atlas</span>, idents <span class="op">=</span> <span class="va">idents</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="preparing-data-for-comparison">Preparing data for comparison<a class="anchor" aria-label="anchor" href="#preparing-data-for-comparison"></a>
</h2>
<p>Now we can pull all of the pathways that we want to compare. For
this, we’re using a csv file that contains a combination of canonical
pathways, gene ontology pathways, and regulatory pathways from <a href="https://www.gsea-msigdb.org/gsea/msigdb" class="external-link">MSigDB</a>, and you can
find this file <a href="https://github.com/jackbibby1/SCPA/blob/main/gene_sets/" class="external-link">here</a>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pathways</span> <span class="op">&lt;-</span> <span class="st">"h_k_r_go_pid_reg_wik.csv"</span></span></code></pre></div>
<p>And let’s create an object with our cell types that we want to
compare, and split the object by disease status.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cell_types</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">blood_atlas</span><span class="op">$</span><span class="va">cell.type.fine</span><span class="op">)</span></span>
<span><span class="va">blood_atlas</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SplitObject.html" class="external-link">SplitObject</a></span><span class="op">(</span><span class="va">blood_atlas</span>, split.by <span class="op">=</span> <span class="st">"Status"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="scpa-comparison">SCPA comparison<a class="anchor" aria-label="anchor" href="#scpa-comparison"></a>
</h2>
<p>Now we’ve formatted everything properly, we just need to loop SCPA
over all the cell types in the dataset. Here we’re using
<code>seurat_extract</code> to pull expression matrices from each cell
type, and then using these as the input to
<code>compare_pathways</code>. We’re using a load of pathways and cell
types here, so this may take a while. To make the downstream analysis
easier, we’re only going to keep the Pathway and qval columns from the
SCPA output, and then add the cell type to the column names, so we can
keep track of the qvals for each cell type.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">scpa_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="va">cell_types</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="va">healthy</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/seurat_extract.html">seurat_extract</a></span><span class="op">(</span><span class="va">blood_atlas</span><span class="op">$</span><span class="va">Healthy</span>, </span>
<span>                            meta1 <span class="op">=</span> <span class="st">"cell.type.fine"</span>, value_meta1 <span class="op">=</span> <span class="va">i</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="va">covid</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/seurat_extract.html">seurat_extract</a></span><span class="op">(</span><span class="va">blood_atlas</span><span class="op">$</span><span class="va">COVID</span>, </span>
<span>                          meta1 <span class="op">=</span> <span class="st">"cell.type.fine"</span>, value_meta1 <span class="op">=</span> <span class="va">i</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"comparing"</span>, <span class="va">i</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">scpa_out</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compare_pathways.html">compare_pathways</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">healthy</span>, <span class="va">covid</span><span class="op">)</span>, <span class="va">pathways</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Pathway</span>, <span class="va">qval</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://magrittr.tidyverse.org/reference/aliases.html" class="external-link">set_colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Pathway"</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">i</span>, <span class="st">"qval"</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># For faster analysis with parallel processing, use 'parallel = TRUE' and 'cores = x' arguments</span></span>
<span>  </span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="plotting-the-output">Plotting the output<a class="anchor" aria-label="anchor" href="#plotting-the-output"></a>
</h2>
<details><summary><strong>Click here for the data wrangling for plotting</strong>
</summary><p>Let’s combine the results from all cell types, and just take pathways
with a qval of &gt; 2 in any comparison</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">scpa_out</span> <span class="op">&lt;-</span> <span class="va">scpa_out</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://purrr.tidyverse.org/reference/reduce.html" class="external-link">reduce</a></span><span class="op">(</span><span class="va">full_join</span>, by <span class="op">=</span> <span class="st">"Pathway"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://magrittr.tidyverse.org/reference/aliases.html" class="external-link">set_colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">" "</span>, replacement <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Pathway"</span>, <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"_qval"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter_all.html" class="external-link">filter_all</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/all_vars.html" class="external-link">any_vars</a></span><span class="op">(</span><span class="va">.</span> <span class="op">&gt;</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"Pathway"</span><span class="op">)</span></span></code></pre></div>
<p>We can then take pathways that we want to highlight in the final
plot</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">blood_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HALLMARK_TNFA_SIGNALING_VIA_NFKB"</span>, <span class="st">"HALLMARK_INFLAMMATORY_RESPONSE"</span>,</span>
<span>                 <span class="st">"HALLMARK_COMPLEMENT"</span>, <span class="st">"HALLMARK_IL6_JAK_STAT3_SIGNALING"</span>,</span>
<span>                 <span class="st">"HALLMARK_IL2_STAT5_SIGNALING"</span>, <span class="st">"HALLMARK_INTERFERON_GAMMA_RESPONSE"</span>,</span>
<span>                 <span class="st">"HALLMARK_MTORC1_SIGNALING"</span>, <span class="st">"HALLMARK_INTERFERON_ALPHA_RESPONSE"</span>,</span>
<span>                 <span class="st">"HALLMARK_MYC_TARGETS_V1"</span>, <span class="st">"HALLMARK_OXIDATIVE_PHOSPHORYLATION"</span><span class="op">)</span></span></code></pre></div>
<p>And create row annotations for the heatmap to highlight these
pathways, and colour scale</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">position</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">scpa_out</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">blood_paths</span><span class="op">)</span></span>
<span><span class="va">row_an</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/rowAnnotation.html" class="external-link">rowAnnotation</a></span><span class="op">(</span>Genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/anno_mark.html" class="external-link">anno_mark</a></span><span class="op">(</span>at <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">scpa_out</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">blood_paths</span><span class="op">)</span>,</span>
<span>                                          labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">scpa_out</span><span class="op">)</span><span class="op">[</span><span class="va">position</span><span class="op">]</span>,</span>
<span>                                          labels_gp <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/gpar.html" class="external-link">gpar</a></span><span class="op">(</span>fontsize <span class="op">=</span> <span class="fl">7</span><span class="op">)</span>,</span>
<span>                                          link_width <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">2.5</span>, <span class="st">"mm"</span><span class="op">)</span>,</span>
<span>                                          padding <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">1</span>, <span class="st">"mm"</span><span class="op">)</span>,</span>
<span>                                          link_gp <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/gpar.html" class="external-link">gpar</a></span><span class="op">(</span>lwd <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">col_hm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/circlize/man/colorRamp2.html" class="external-link">colorRamp2</a></span><span class="op">(</span>colors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"white"</span>, <span class="st">"red"</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">3</span>, <span class="fl">6</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</details><p><br></p>
<p>Now we have all the results, we can plot a heatmap of the qvals to
get an idea of any broad patterns. Interestingly, we see a large
dysregulation of pathways in CD14<sup>+</sup> monocytes in the
peripheral blood of COVID-19 patients, suggesting that these cell show
the greatest deviation from CD14<sup>+</sup> monocytes from healthy
donors. These pathways include well defined immunological signatures in
response to viral infection, including interferon response pathways, and
complement cascades. We’ve just chosen to highlight a few on the heatmap
below.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span><span class="va">scpa_out</span>,</span>
<span>        col <span class="op">=</span> <span class="va">col_hm</span>,</span>
<span>        name <span class="op">=</span> <span class="st">"Qval"</span>,</span>
<span>        show_row_names <span class="op">=</span> <span class="cn">F</span>,</span>
<span>        right_annotation <span class="op">=</span> <span class="va">row_an</span>,</span>
<span>        column_names_gp <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/gpar.html" class="external-link">gpar</a></span><span class="op">(</span>fontsize <span class="op">=</span> <span class="fl">8</span><span class="op">)</span>,</span>
<span>        border <span class="op">=</span> <span class="cn">T</span>,</span>
<span>        column_km <span class="op">=</span> <span class="fl">3</span>,</span>
<span>        row_km <span class="op">=</span> <span class="fl">3</span>,</span>
<span>        column_labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CS B"</span>, <span class="st">"IgG B"</span>, <span class="st">"IgA B"</span>, <span class="st">"CD14 Mono"</span>, <span class="st">"mCD8 T"</span>, <span class="st">"mCD4 T"</span>,</span>
<span>                          <span class="st">"nCD4 T"</span>, <span class="st">"B"</span>, <span class="st">"NK"</span>, <span class="st">"Neut"</span>, <span class="st">"CD16 Mono"</span>, <span class="st">"gdT"</span>, <span class="st">"pDC"</span>, <span class="st">"Eosin"</span>, <span class="st">"DC"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="../reference/figures/covid_heatmap.png" style="width:75.0%"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Jack Bibby, Divyansh Agarwal.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
